rusk-wallet/src/clients.rs

```diff
@@ -248,133 +130,126 @@ impl StateStore {
+                .collect();
+
+            try_input_notes(inputs?, target)?
+                .into_iter()
+                .map(|(note, _)| {
+                    let opening = self.fetch_opening(&note)?;
+
+                    Ok((note, opening))
+                })
+                .collect()
+        } else {
+            Err(Error::AddressNotOwned)
+        }
+    }
+
+    pub(crate) fn fetch_notes(
         &self,
-        vk: &ViewKey,
-    ) -> Result<Vec<EnrichedNote>, Self::Error> {
-        let psk = vk.public_spend_key();
-        let state = self.inner.lock().unwrap();
-
-        Ok(state
-            .cache
-            .notes(&psk)?
+        pk: &PhoenixPublicKey,
+    ) -> Result<Vec<NoteData>, Error> {
+        self.cache()
+            .notes(pk)?
             .into_iter()
-            .map(|data| (data.note, data.height))
-            .collect())
+            .map(|data| {
+                Ok(NoteData {
+                    note: data.note,
+                    height: data.height,
+                })
+            })
+            .collect()
     }
 
     /// Fetch the current anchor of the state.
-    fn fetch_anchor(&self) -> Result<BlsScalar, Self::Error> {
-        let state = self.inner.lock().unwrap();
-
-        self.status("Fetching anchor...");
+    pub(crate) fn fetch_anchor(&self) -> Result<BlsScalar, Error> {
+        let status = self.status;
+        status("Fetching anchor...");
 
-        let anchor = state
+        let anchor = self
             .client
             .contract_query::<(), 0>(TRANSFER_CONTRACT, "root", &())
             .wait()?;
-        self.status("Anchor received!");
+        status("Anchor received!");
         let anchor = rkyv::from_bytes(&anchor).map_err(|_| Error::Rkyv)?;
         Ok(anchor)
     }
 
-    /// Asks the node to return the nullifiers that already exist from the given
-    /// nullifiers.
-    fn fetch_existing_nullifiers(
-        &self,
-        _nullifiers: &[BlsScalar],
-    ) -> Result<Vec<BlsScalar>, Self::Error> {
-        Ok(vec![])
-    }
-
-    /// Queries the node to find the opening for a specific note.
-    fn fetch_opening(
-        &self,
-        note: &Note,
-    ) -> Result<PoseidonOpening<(), POSEIDON_TREE_DEPTH, 4>, Self::Error> {
-        let state = self.inner.lock().unwrap();
-
-        self.status("Fetching opening notes...");
-
-        let data = state
-            .client
-            .contract_query::<_, 1024>(TRANSFER_CONTRACT, "opening", note.pos())
-            .wait()?;
-
-        self.status("Opening notes received!");
-
-        let branch = rkyv::from_bytes(&data).map_err(|_| Error::Rkyv)?;
-        Ok(branch)
-    }
-
     /// Queries the node for the amount staked by a key.
-    fn fetch_stake(&self, pk: &PublicKey) -> Result<StakeInfo, Self::Error> {
-        let state = self.inner.lock().unwrap();
-
-        self.status("Fetching stake...");
+    pub(crate) fn fetch_stake(
+        &self,
+        pk: &PhoenixPublicKey,
```

Comment:
This need to a AccountPublicKey, that's why you're not getting a stake

Reply:
Changed to account Public Key